### PR TITLE
Add gitops version subcommand

### PR DESCRIPTION
**Pull request by wtam2018**
**What type of PR is this?**
/kind feature

**What does this PR do / why we need it**:

This PR implements a "version" sub-command for gitops command.   What it does is that it outputs the version of gitops CLI with the following information so that the user can easily find out what source revision the gitops CLI was built from.

```<tag>-<n-commits>-g<commit-sha>-<dirty>```

<tag> is the tag if tag has been created
<n-commits> is the number of commits since the tag if tag has been created.
g - is for "git" if tag has been created.
<commit-sha> is the short commit sha
<dirty> is shown if the command binary was built with local file changes.

E.g.
```
gitops version v0.0.9-1-g7fa007f-dirty
```

**Have you updated the necessary documentation?**

See https://github.com/rhd-gitops-example/docs

[ ] Updated

**Which issue(s) this PR fixes**:

Fixes #?
https://issues.redhat.com/browse/GITOPS-310

**How to test changes / Special notes to the reviewer**:
Manually tested.
```shell
$ make clean
$ make bin
go build -i -v -o gitops -ldflags="-s -w -X github.com/rhd-gitops-example/gitops-cli/pkg/cmd/version.Version=003abdb" cmd/gitops-cli/gitops.go 
$ ./gitops version
gitops version 003abdb
```

### CODE REVIEW

File: pkg/cmd/root.go [hunk 1]
```diff
@@ -6,19 +6,20 @@ import (
...
 
 var (
 	gitopsLong = "CLI tool to scaffold your GitOps repository"
-	fullName   = "gitops-cli"
+	fullName   = "gitops"
 )
 
 func makeRootCmd() *cobra.Command {
 	rootCmd := &cobra.Command{
-		Use:   "gitops-cli",
```
Comment (bigkevmcd) [inline pkg/cmd/root.go:20]: You probably want to rename `cmd/gitops-cli` with this change.

File: pkg/cmd/version/version.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package version
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// RecommendedCommandName is the recommended environment command name.
```
Comment (bigkevmcd) [inline pkg/cmd/version/version.go:9]: Not sure what `environment` means in this context?

File: pkg/cmd/version/version.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const RecommendedCommandName = "version"
+
+var Version string
+
+// NewCmd creates a new environment command
```
Comment (bigkevmcd) [inline pkg/cmd/version/version.go:14]: Or here?

These could probably be `version` ?

File: pkg/cmd/version/version.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+func NewCmd(name, fullName string) *cobra.Command {
+	return &cobra.Command{
+		Use:   name,
+		Short: "Get gitops command version",
```
Comment (bigkevmcd) [inline pkg/cmd/version/version.go:18]: How about...

```Print the version information``` as it's not really "getting" anything?

**Review by bigkevmcd — CHANGES_REQUESTED** [review]

File: pkg/cmd/root.go (hunk 1 above)
Comment (wtam2018) [reply pkg/cmd/root.go:20]: renamed root.go to gitops.go

File: pkg/cmd/version/version.go (hunk 2 above)
Comment (wtam2018) [reply pkg/cmd/version/version.go]: removed "environment" (bad copy and past)

File: pkg/cmd/version/version.go (hunk 3 above)
Comment (wtam2018) [reply pkg/cmd/version/version.go]: removed "environment" (bad copy and past)

File: pkg/cmd/version/version.go (hunk 4 above)
Comment (wtam2018) [reply pkg/cmd/version/version.go]: Replaced text as suggested.

**Review by bigkevmcd — APPROVED** [review]

**Pull request merged**